src/Features/CSharp/Portable/MakeLocalFunctionStatic/MakeLocalFunctionStaticHelper.cs

```diff
@@ -0,0 +1,147 @@
+        {
+            var dataFlow = semanticModel.AnalyzeDataFlow(localFunction);
+            captures = dataFlow.CapturedInside;
+
+            return dataFlow.Succeeded;
+        }
+
+        public static async Task<Document> MakeLocalFunctionStaticAsync(
+            LocalFunctionStatementSyntax localFunction,
+            ImmutableArray<ISymbol> captures,
+            SemanticModel semanticModel,
+            Document document,
+            CancellationToken cancellationToken)
+        {
+            var root = (await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false))!;
+
+            var localFunctionSymbol = semanticModel.GetDeclaredSymbol(localFunction, cancellationToken);
+            var parameters = CreateParameterSymbols(captures);
+
+            // Finds all the call sites of the local function
+            var referencedSymbols = await SymbolFinder.FindReferencesAsync(
+                localFunctionSymbol, document.Project.Solution, cancellationToken).ConfigureAwait(false);
+
+            // Now we need to find all the refereces to the local function that we might need to fix.
+            var shouldWarn = false;
+            var invocationReferences = new List<InvocationExpressionSyntax>();
+
+            foreach (var referencedSymbol in referencedSymbols)
+            {
+                foreach (var location in referencedSymbol.Locations)
+                {
+                    // Since this is a local function, all reference must be in the same tree.
+                    var referenceNode = root.FindNode(location.Location.SourceSpan);
+                    if (!(referenceNode is IdentifierNameSyntax identifierNode))
+                    {
+                        continue;
+                    }
+
+                    if (identifierNode.Parent is InvocationExpressionSyntax invocation)
+                    {
+                        invocationReferences.Add(invocation);
+                    }
+                    else
+                    {
+                        // We won't be able to fix non-invocation references, 
+                        // e.g. creating a delegate. 
+                        shouldWarn = true;
+                    }
+                }
+            }
+
+            var nodeToTrack = new List<SyntaxNode>(invocationReferences) { localFunction };
+            root = root.TrackNodes(nodeToTrack);
+
+            // Fix all invocations by passing in additional arguments.
+            foreach (var invocation in invocationReferences.OrderByDescending(n => n.Span.Start))
+            {
+                var currentInvocation = root.GetCurrentNode(invocation);
+
+                var seenNamedArgument = currentInvocation.ArgumentList.Arguments.Any(a => a.NameColon != null);
+                var seenDefaultArgumentValue = currentInvocation.ArgumentList.Arguments.Count < localFunction.ParameterList.Parameters.Count;
```

Comment:
1. tests for theses cases?
2. want to be really cute and check if 'cancellationToken' is the last parameter and preserve that? :)